### PR TITLE
Develop plugin declaration inspections

### PR DESCRIPTION
**Pull request by Iamwade**
**Description** (*)
Added inspection to the di.xml file for plugin tag.
The inspection is performed for plugin attribute "type".
If the class in the specified path for these attributes does not exist or empty, then the corresponding warning will be displayed
image
image
image

Inspection for the class inside the  main "type" tag was done in the issue https://github.com/magento/magento2-phpstorm-plugin/pull/582

**Fixed Issues (if relevant)**
1. Fixes magento/magento2-phpstorm-plugin#579

**Contribution checklist** (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with integration/functional tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: tests/com/magento/idea/magento2plugin/inspections/xml/PluginAttrTypeInspectionTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        final String forAttrIsEmptyMessage =  inspectionBundle.message(
+                ARGUMENT_VALUE_IS_EMPTY,
+                ModuleDiXml.TYPE_ATTR
+        );
+
+        assertHasHighlighting(forAttrIsEmptyMessage);
+    }
+
+    /**
+     * Test for an no error for the "type" attribute because this class exists.
```
[the remark this answers is not in the document]
Comment (Iamwade) [reply tests/com/magento/idea/magento2plugin/inspections/xml/PluginAttrTypeInspectionTest.java:43]: fixed

File: resources/META-INF/plugin.xml [hunk 2]
```diff
@@ -229,6 +229,13 @@
                          enabledByDefault="true" level="WARNING"
                          implementationClass="com.magento.idea.magento2plugin.inspections.xml.PreferenceDeclarationInspection"/>
 
+        <localInspection language="XML" groupPath="XML"
+                         shortName="PluginAttrTypeInspection"
+                         bundle="magento2.inspection" key="inspection.displayName.PluginAttrTypeInspection"
+                         groupBundle="magento2.inspection" groupKey="inspection.group.name"
+                         enabledByDefault="true" level="WARNING"
+                         implementationClass="com.magento.idea.magento2plugin.inspections.xml.PluginAttrTypeInspection"/>
```
[the remark this answers is not in the document]
Comment (Iamwade) [reply resources/META-INF/plugin.xml:237]: fixed

File: src/com/magento/idea/magento2plugin/inspections/xml/PluginAttrTypeInspection.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+import com.intellij.psi.xml.XmlAttribute;
+import com.intellij.psi.xml.XmlTag;
+import com.magento.idea.magento2plugin.bundles.InspectionBundle;
+import com.magento.idea.magento2plugin.inspections.validator.InspectionValidator;
+import com.magento.idea.magento2plugin.inspections.validator.NotEmptyValidator;
+import com.magento.idea.magento2plugin.inspections.validator.PhpClassExistenceValidator;
+import com.magento.idea.magento2plugin.magento.files.ModuleDiXml;
+import org.jetbrains.annotations.NotNull;
+
+public class PluginAttrTypeInspection extends XmlSuppressableInspectionTool {
```
[the remark this answers is not in the document]
Comment (Iamwade) [reply src/com/magento/idea/magento2plugin/inspections/xml/PluginAttrTypeInspection.java:24]: fixed

**Comment by Iamwade** [comment]
@VitaliyBoyko  fixed, Thank you.

**Review by VitaliyBoyko — APPROVED** [review]

**Pull request merged**